### ISSUE TITLE
Failed to get coverage result: Unable to read beyond the end of the stream

### ISSUE THREAD

**Issue by caghand** [coverlet-coverage/coverlet#972]
This is a follow-up to issue #718.
We are seeing this issue *even* when using the DataCollector integration ("XPlat Code Coverage").
But it only happens for a .csproj file that has 6500 tests. :)
And it happens once every 5 runs, approximately.

Our package versions:
- Microsoft.NET.Test.Sdk: 16.7.1
- coverlet.collector: 1.3.0

**Comment by fahadabdulaziz** [comment]
I also get this error:
`
2020-10-18T14:30:16.9285616Z Test Run Successful.
2020-10-18T14:30:16.9285973Z Total tests: 446
2020-10-18T14:30:16.9286228Z      Passed: 446
2020-10-18T14:30:16.9288973Z  Total time: 7.6951 Seconds
2020-10-18T14:30:16.9447417Z 
2020-10-18T14:30:16.9447938Z Calculating coverage result...
2020-10-18T14:30:17.3136217Z C:\WINDOWS\ServiceProfiles\NetworkService\.nuget\packages\coverlet.msbuild\2.9.0\build\coverlet.msbuild.targets(60,5): error : Unable to read beyond the end of the stream. [C:\agent\_work\1\s\Tests\ProjectName\ProjectName.csproj]
2020-10-18T14:30:17.3138172Z C:\WINDOWS\ServiceProfiles\NetworkService\.nuget\packages\coverlet.msbuild\2.9.0\build\coverlet.msbuild.targets(60,5): error :    at System.IO.BinaryReader.InternalRead(Int32 numBytes) [C:\agent\_work\1\s\Tests\ProjectName\ProjectName.csproj]
2020-10-18T14:30:17.3139390Z C:\WINDOWS\ServiceProfiles\NetworkService\.nuget\packages\coverlet.msbuild\2.9.0\build\coverlet.msbuild.targets(60,5): error :    at Coverlet.Core.Coverage.CalculateCoverage() in /_/src/coverlet.core/Coverage.cs:line 359 [C:\agent\_work\1\s\Tests\ProjectName\ProjectName.csproj]
2020-10-18T14:30:17.3140793Z C:\WINDOWS\ServiceProfiles\NetworkService\.nuget\packages\coverlet.msbuild\2.9.0\build\coverlet.msbuild.targets(60,5): error :    at Coverlet.Core.Coverage.GetCoverageResult() in /_/src/coverlet.core/Coverage.cs:line 141 [C:\agent\_work\1\s\Tests\ProjectName\ProjectName.csproj]
2020-10-18T14:30:17.3142127Z C:\WINDOWS\ServiceProfiles\NetworkService\.nuget\packages\coverlet.msbuild\2.9.0\build\coverlet.msbuild.targets(60,5): error :    at Coverlet.MSbuild.Tasks.CoverageResultTask.Execute() in /_/src/coverlet.msbuild.tasks/CoverageResultTask.cs:line 112 [C:\agent\_work\1\s\Tests\ProjectName.csproj]
`
* coverlet.collector: 1.3.0
* coverlet.msbuild: 2.9.0
* Microsoft.NET.Test.Sdk: 16.7.1

**Comment by caghand** [comment]
@fahadabdulaziz, your situation is a known issue in the *msbuild* integration, addressed in issue #718.  *This* issue is dedicated to investigating the situation where it happens in the *DataCollector* integration (*without* any msbuild integration).

**Comment by fahadabdulaziz** [comment]
@caghand oh thanks for clarifying that, and sorry for the interruption :-)

**Comment by MarcoRossignoli** [comment]
@fahadabdulaziz is your project .NET Framework or .NET Core?

**Comment by caghand** [comment]
Hi @MarcoRossignoli, our project is in .NET Core 3.1.
*But* we are *not* passing "XPlat Code Coverage" on the command line.  We are passing it via a ".runsettings" file.
Our command line looks like this:
`dotnet test <path_to_csproj_file> --no-build --verbosity normal --configuration "$(BuildConfiguration)" /property:Platform="$(BuildPlatform)" --settings test.runsettings`
And test.runsettings contains a DataCollector element with friendlyName="XPlat Code Coverage".
Could this problem be happening because we are *not* passing "XPlat Code Coverage" on the command line?

**Comment by MarcoRossignoli** [comment]
> Could this problem be happening because we are not passing "XPlat Code Coverage" on the command line?

Yep without passing it to command line in process collector won't be injected(and you'll hit same issue as msbuild), we use runsettings only to pass parameters to coverlet.
This is second time I see this misuse, can you file an issue on https://github.com/microsoft/vstest so I can understand how to fix?

It's on guide https://github.com/coverlet-coverage/coverlet/blob/master/Documentation/VSTestIntegration.md but I think it isn't clear enough 😕  , let me know if we can improve it, or feel free to open a PR to "highlight" this fact.

**Comment by caghand** [comment]
Thanks for the info!  I have filed an issue about it here: https://github.com/microsoft/vstest/issues/2599 (I hope I explained it correctly there! :))
I think your documentation is pretty clear.  And my first impression was just like you intended: The ".runsettings" file contains DataCollector settings, and the command-line contains which DataCollectors to activate.  But then I noticed an unexpected behavior: I had an old "Code Coverage" element left over in my ".runsettings" file, and this caused `dotnet test` to collect legacy ".coverage" reports *as well as* Cobertura reports.  And then I thought, "Hey, it looks like the '.runsettings' file actually contains the list of DataCollectors to *activate*, and the command-line parameter must be redundant.  Let me avoid repeating the same 'magic string' in two places."  And of course, this was a wrong conclusion.
After adding back the command-line parameter, we haven't seen this problem (so far), so I am closing this issue! :)
